recipes/protoc/3.9.x/test_package/CMakeLists.txt

```diff
@@ -1,21 +1,19 @@
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.1)
 project(test_package CXX)
 
-set(CMAKE_VERBOSE_MAKEFILE ON)
-
-# Findprotoc.cmake generated by Conan does not give a hint where protoc is installed.
 include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
 conan_basic_setup()
 
-find_package(protoc)
+find_package(Protobuf REQUIRED)
+find_package(protoc REQUIRED)
 
 if (NOT protobuf_MODULE_COMPATIBLE)
     message(FATAL_ERROR "protobuf_MODULE_COMPATIBLE is mandatory")
 endif()
 
-PROTOBUF_GENERATE_CPP(PROTO_SRCS PROTO_HDRS addressbook.proto)
+add_executable(${PROJECT_NAME} test_package.cpp addressbook.proto)
```

Comment:
add_executable should not be required. Only protobuf_generate_cpp should be enough

Reply:
Why did you change the add_executable order to the beginning?

Reply:
It now uses the `TARGET` property of the `protobuf_generate_cpp` function.
The function will scan the TARGET for `*.proto` sources and run protoc on them.
It looks much nicer then fiddling with cmake variables imho.